### PR TITLE
feat: add an option for pre-built components

### PR DESCRIPTION
**Pull request by oubenruing**
To avoid infinite reloads caused by the introduction of dynamic components, add an option to make vite add the specified UI component library to optimizeDeps.include in the 'config' lifecycle for all prebuilds. #235 #255 
（This PR is for antdv resolver and vant resolver）

### CODE REVIEW

File: src/core/resolvers/vant.ts [hunk 1]
```diff
@@ -10,6 +10,103 @@ export interface VantResolverOptions {
...
+  'Toast',
+  'TreeSelect',
+  'Uploader',
+]
+
+const prefix = 'Van'
+
+let vanNames: Set<string>
+
+function genAntdNames(primitiveNames: string[]): void {
```
Comment (nabaonan) [inline src/core/resolvers/vant.ts:105]: i think it is not a good function name

Comment (nabaonan) [reply src/core/resolvers/vant.ts:105]: maybe you can rename it  genVantNames

Comment (oubenruing) [reply src/core/resolvers/vant.ts]: Thank you for the reminder

**Review by oubenruing — COMMENTED** [review]
Name has been changed.

**Comment by antfu** [comment]
I don't really like this approach and would think making optimization for components is out-of-scope of this project. I feel it's better to either be handled by another plugin, provide on the user side, or for libraries like antdv and vant to ship the ESM version directly.

**Comment by oubenruing** [comment]
> I don't really like this approach and would think making deps optimization for components is out-of-scope of this project. I feel it's better to either be handled by another plugin, provide on the user side, or for libraries like antdv and vant to ship the ESM version directly.

I actually added this option to fix that, after unplugin-vue-components imported components from the UI library (I use antdv ESM) on demand, vite always found a new **style dependency** (css or less in antdv) and repeatedly reloaded the page, causing it to load very slowly.

``` shell
[vite] new dependencies found: ant-design-vue/es/divider/style, ant-design-vue/es/popover/style, updating...
[vite] ✨ dependencies updated, reloading page...
//......
[vite] new dependencies found: ant-design-vue/es/spin/style, updating...
[vite] ✨ dependencies updated, reloading page...

```

I actually have two solutions

1. Setting importStyle to false in the options of resolver can avoid this. But this requires importing the full style file.
2.  I noticed this project [vite-plugin-optimize-persist](https://github.com/antfu/vite-plugin-optimize-persist), which is awsome. But the first load still consumes a lot of time. That's why I created this PR, hoping to solve the Pre-Bundling of on-demand import on the first load to improve the experience during the development phase. (It seems pointless to make a separate plugin to set up pre-bundling without using unplugin-vue-components).

Anyway, thanks for your comments! If you still feel that this PR is not suitable, please close it.

Best regards.

**Pull request closed without merging**